tests/xrt/perf_IOPS/xrt_api_iops.cpp

```diff
@@ -59,29 +90,127 @@ int testSingleThread(const xrt::device& device, const xrt::uuid& uuid)
   std::cout << "Allocated commands, expect " << expected_cmds << ", created " << cmds.size() << std::endl;
 
   for (auto num_cmds : cmds_per_run) {
-    double duration = runTest(cmds, num_cmds);
+    double duration = runTest(cmds, num_cmds, arg[0]);
     std::cout << "Commands: " << std::setw(7) << num_cmds
-              << " iops: " << (num_cmds * 1000.0 * 1000.0 / duration)
-              << std::endl;
+      << " iops: " << (num_cmds * 1000.0 * 1000.0 / duration)
+      << std::endl;
   }
 
   return 0;
 }
 
+void runTestThread(xrt::device &device, xrt::kernel& kernel, arg_t &arg)
+{
+  std::vector<xrt::run> cmds;
+
+  for (int i = 0; i < arg.queueLength; i++) {
+    auto run = xrt::run(kernel);
+    run.set_arg(0, xrt::bo(device, 20, kernel.group_id(0)));
+    cmds.push_back(std::move(run));
+  }
+  barrier.wait();
+
+  double duration = runTest(cmds, arg.total, arg);
+
+  barrier.wait();
+
+}
+
+int testMultiThreads(xrt::device& device, xrt::uuid& uuid,
+        int threadNumber, int queueLength, unsigned int total)
+{
+    std::thread threads[threadNumber];
+    std::vector<arg_t> arg(threadNumber);
+
+    auto hello = xrt::kernel(device, uuid.get(), "hello");
+
+    barrier.init(threadNumber + 1);
+
+    for (int i = 0; i < threadNumber; i++) {
+      arg[i].thread_id = i;
+      arg[i].queueLength = queueLength;
+      arg[i].total = total;
+      threads[i] = std::move(std::thread(runTestThread, std::ref(device), std::ref(hello), std::ref(arg[i])));
```

Comment:
We are no longer in 1998... :-)
```suggestion
      threads[i] = std::thread { [&] { runTestThread(device, hello, arg[i]) } };
```
Perhaps
```c++
threads[i] = { [&] { runTestThread(device, hello, arg[i]) } };
```
could work too?

Reply:
Thanks @keryell . Use lambda function with std::thread is a great idea. Learn something new today.

Your suggestion missed that 'i' is local variable in the loop. Below line works.
`threads[i] = std::thread([&](int i){ runTestThread(device, hello, arg[i]); }, i);`

Reply:
Yes, this is a new language called... modern C++! ;-) I will invite you to #cpp on Slack.
Too bad C++ is not well known at Xilinx...
Anyway, I missed that if we capture `i` per reference we will be in trouble... :-(
Good to see you figured out a clever way to handle it!
But there is a simpler and official way to do this according to
https://en.cppreference.com/w/cpp/language/lambda
Could you try for me:
```c++
threads[i] = { [&, i] { runTestThread(device, hello, arg[i]); } };
```